lib/modules/datasource/rubygems/index.ts

```diff
@@ -34,16 +35,26 @@ export class RubyGemsDatasource extends Datasource {
     key: ({ registryUrl, packageName }: GetReleasesConfig) =>
       `${registryUrl}/${packageName}`,
   })
-  getReleases({
+  async getReleases({
     packageName,
     registryUrl,
   }: GetReleasesConfig): Promise<ReleaseResult | null> {
     if (parseUrl(registryUrl)?.hostname === 'rubygems.org') {
       return this.rubyGemsOrgDatasource.getReleases({ packageName });
     }
-    return this.internalRubyGemsDatasource.getReleases({
-      packageName,
-      registryUrl,
-    });
+    try {
+      return await this.internalRubyGemsDatasource.getReleases({
+        packageName,
+        registryUrl,
+      });
+    } catch (error) {
+      if (error instanceof HttpError && error.response?.statusCode === 404) {
```

Comment:
This could occur for either of these scenarios, right?

1. The registry doesn't support the API at all, or
2. The particular dependency is not found

Also, for (1). I think we're going to get X failed HTTP requests if we have X dependencies. Ideally we could somehow "probe" a gem registry once to determine what it supports and then know in future whether to use the versions interface or the API.

ALSO we should keep in mind that we in future want to support a hybrid approach for rubygems - use /versions as much as possible for detecting releases but lightly use the API for fetching metadata. Similar to what we've done with crates.io